src/Compilers/Core/Portable/CommandLine/AnalyzerConfig.cs

```diff
@@ -75,19 +75,32 @@ public sealed partial class AnalyzerConfig
 
         internal ImmutableArray<Section> NamedSections { get; }
 
+        /// <summary>
+        /// Gets the concatenation of the <see cref="GlobalSection" /> and <see cref="NamedSections" />
+        /// </summary>
+        internal ImmutableArray<Section> AllSections { get; }
```

Comment:
Why did you choose to expose this an an `ImmutableArray` vs. `IEnumerable`? Had a small worry about the extra allocations this will introduce.

Reply:
Only to match the existing properties. Given that we have to concat the two immutable arrays, is there an allocation free way of doing it? `Concat` ? 

---
In reply to: [421674518](https://github.com/dotnet/roslyn/pull/43889#discussion_r421674518) [](ancestors = 421674518)

Reply:
Ah looks like its `Prepend` i'm looking for.

---
In reply to: [421695250](https://github.com/dotnet/roslyn/pull/43889#discussion_r421695250) [](ancestors = 421695250,421674518)